artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/FederatedQueueConsumerImpl.java

```diff
@@ -129,21 +134,23 @@ private void connect() throws Exception {
                   throw new ActiveMQNonExistentQueueException("Queue " + key.getQueueName() + " does not exist on remote");
                }
             }
-         }
-      } catch (Exception e) {
-         try {
-            if (clientSessionFactory != null) {
-               clientSessionFactory.cleanup();
+         } catch (Exception e) {
+            try {
+               if (clientSessionFactory != null) {
+                  clientSessionFactory.cleanup();
+               }
+               disconnect();
+            } catch (ActiveMQException ignored) {
             }
-            disconnect();
-         } catch (ActiveMQException ignored) {
+            throw e;
```

Comment:
Not necessarily the spot to do it (perhaps in the task where it retries), but the easiest to comment in this PR. It seems like it might be good to have at least a trace log to indicate it has failed and is going to try connecting again.

Reply:
Logging added in `scheduleConnect`.